### PR TITLE
style: lint docs with markdownlint, introduce GitHub's super-linter (DEV-1924)

### PR DESCRIPTION
**Pull request by jnussbaum**
resolves DEV-1924

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [![PyPI version](https://badge.fury.io/py/dsp-tools.svg)](https://badge.fury.io/py/dsp-tools)
+[![GitHub Super-Linter](https://github.com/dasch-swiss/dsp-tools/workflows/Lint%20Code%20Base/badge.svg)](https://github.com/marketplace/actions/super-linter)
```
Comment (jnussbaum) [inline README.md:2]: This here doesn't seem to work, but I don't know why...

File: .github/workflows/check-pr-title.yml [hunk 2]
```diff
@@ -1,3 +1,5 @@
+---
+
```
Comment (BalduinLandolt) [inline .github/workflows/check-pr-title.yml:2]: is this really needed?

File: .markdownlint.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+    "MD006": false,
+    "MD007": {"start_indented": true, "indent": 2, "start_indent": 2},
```
Comment (BalduinLandolt) [inline .markdownlint.json:3]: I personally don't like `start_indent: 2`

**Review by BalduinLandolt — APPROVED** [review]
I'm not sure how high priority this kind of housekeeping should be, but generally I like well-linted codebases. I added some minor remarks

File: .github/workflows/check-pr-title.yml (hunk 2 above)
Comment (jnussbaum) [reply .github/workflows/check-pr-title.yml:2]: the yaml-linter told me so... (I could of course add a config file for the yaml linter, and tell him that I don't care about the three dashes, but that would be too much effort).

File: .github/workflows/tests-on-push.yml [hunk 4]
```diff
@@ -9,10 +11,12 @@ jobs:
   tests-on-push:
     runs-on: ubuntu-latest
     steps:
+    # preparation
       - name: Checkout source
         uses: actions/checkout@v3
         with:
-          fetch-depth: 1
+          # Full git history is needed to get a proper list of changed files within `super-linter`
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (jnussbaum) [reply .github/workflows/tests-on-push.yml:19]: The docs of super-linter told me it's necessary, so I just believed them. But if I want to lint the entire codebase anyway, it shouldn't be necessary...

File: .markdownlint.json (hunk 3 above)
Comment (jnussbaum) [reply .markdownlint.json:3]: Do you prefer no indentation (=standard markdown)? I know that with this, I'm moving away from standard markdown, which is perhaps not the best idea...

File: .github/workflows/check-pr-title.yml (hunk 2 above)
Comment (BalduinLandolt) [reply .github/workflows/check-pr-title.yml:2]: That feels like tooling working against you...

File: .markdownlint.json (hunk 3 above)
Comment (BalduinLandolt) [reply .markdownlint.json:3]: yes, I would keep it standard where ever possible, unless there is a good reason for it

**Pull request merged**